### ISSUE TITLE
i18n Configuration Breaks /docs Route After CLI Integration

### ISSUE THREAD

**Issue by omercan** [fuma-nama/fumadocs#1183]
### To Reproduce

	1.	Install fuma using the official setup instructions. (pnpm create fumadocs-app)
	2.  Install CLI (pnpm add @fumadocs/cli --save-dev)
	3. Init i18n (pnpm fumadocs init i18n)
	4. change lib/i18n cn to tr
	5. go to http://localhost:3000/tr/docs or http://localhost:3000/en/docs or http://localhost:3000/docs
	6. 
![Screenshot 2024-12-24 at 21 30 08](https://github.com/user-attachments/assets/b994fb70-6a08-4fbf-b0b1-40e4be22a65c)

### Current vs. Expected behavior

I am expecting entering docs subfolder with example md files. after that I will create nwe ones with .tr.mdx lang extension

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:02:45 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T8112
  Available memory (MB): 24576
  Available CPU cores: 8
Binaries:
  Node: 22.12.0
  npm: 10.9.0
  Yarn: N/A
  pnpm: 9.15.0
Relevant Packages:
  next: 15.1.2 // Latest available version is detected (15.1.2).
  eslint-config-next: N/A
  react: 19.0.0
  react-dom: 19.0.0
  typescript: N/A
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Core

**Comment by fuma-nama** [comment]
Hmm Next.js 15 made it async, will update it later

**Comment by omercan** [comment]
is there any quick fix? @fuma-nama

**Comment by omercan** [comment]
still same after the cli changes:

![Screenshot 2024-12-25 at 22 23 03](https://github.com/user-attachments/assets/a109580b-da5f-440d-b786-80b42e03e7fa)

**Comment by fuma-nama** [comment]
Make sure to update to the latest CLI version

**Comment by omercan** [comment]
Still getting error after new clean setup.

![Screenshot 2024-12-26 at 13 49 31](https://github.com/user-attachments/assets/578e4f69-615b-42a9-b90f-566ae497ac78)

**Comment by fuma-nama** [comment]
Hmm maybe you can follow the manual setup guide, it should works

**Comment by omercan** [comment]
This part is not clear on the documentation: 

> Update the usages to your source to include a locale code:
> 
> 
> import { source } from '@/lib/source';
>  
> // get page tree
> source.pageTree[params.lang];
>  
> // get page
> source.getPage(params.slug, params.lang);
>  
> // get pages
> source.getPages(params.lang);
> Note that without providing a locale code, it uses your default locale instead. You can see [Source API](https://fumadocs.vercel.app/docs/headless/source-api) for other usages.

**Comment by fuma-nama** [comment]
Hmm which part, your can update your usage of source by following the guide

**Comment by omercan** [comment]
I couldn’t quite understand where exactly I should add the provided code snippet in the documentation.

**Comment by fuma-nama** [comment]
You can click on the source variable in `source.ts` and find its references, and update the references according to the guide

**Comment by omercan** [comment]
It is really hard to follow manual setup, you cannot understand what is special files mean or find RootLayout I think it is outdated.

**Comment by fuma-nama** [comment]
1. Root layout is very clear, it is required for every Next.js app
2. Next docs also mentioned special files "Finally, ensure all special files inside app/ are nested under app/[lang]", it is a conventional word

**Comment by shiny** [comment]
> It is really hard to follow manual setup, you cannot understand what is special files mean or find RootLayout I think it is outdated.

I agree, especially regarding how the file structure should look after adding `[lang]`.
I'm not quite sure about it.

**Comment by fuma-nama** [comment]
Yeah I agree, for Fumadocs, you have to first migrate your Next.js to i18n routing, then configure Fumadocs. The former is handled by Next.js which we cannot change, the latter is already the simplest, only need to add a provider and pass an option to ‘loader()`.

The Next.js layer is hard to change, the only thing I can improve is the guide, but it's neither easy because Next.js's official i18n guide is also difficult to follow, it requires many refactors (e.g. moving files), and beginners may not be able to catch up.
